go/mysql/flavor_mysql.go

```diff
@@ -121,6 +121,11 @@ func (mysqlFlavor) status(c *Conn) (SlaveStatus, error) {
 	if err != nil {
 		return SlaveStatus{}, vterrors.Wrapf(err, "SlaveStatus can't parse MySQL 5.6 GTID (Executed_Gtid_Set: %#v)", resultMap["Executed_Gtid_Set"])
 	}
+	status.RelayLogPosition = &Position{}
+	status.RelayLogPosition.GTIDSet, err = parseMysql56GTIDSet(resultMap["Retrieved_Gtid_Set"])
```

Comment:
Although the Position struct just consists of a single GTIDSet value, it represents a subtly different concept than GTIDSet. A Position value should specifically contain the set of all GTIDs from the beginning of time (when a shard was initialized) until some particular point in time. All Position values are GTIDSets, but not all GTIDSets are Positions.

Since `Retrieved_Gtid_Set` only tells you what's been retrieved since the last time the relay log was reset, we need to take the union of the retrieved set with `status.Position.GTIDSet` to get a true Position value for the relay log.

Reply:
In this case, would we only need to compare the relay log position, since it will already be the union at that point?

Reply:
Done, and now the emergency reparent code depends on a `decodePosition` helper that will preferably return the RelayLogPosition, if one exists and otherwise falls back to returning the `Position`. All code paths have been updated to depend on this helper for decoding.